offline/QA/Jet/photonjetskinematics.cc

```diff
@@ -0,0 +1,184 @@
+//____________________________________________________________________________..
+
+#include "photonjetskinematics.h"
+#include <fun4all/PHTFileServer.h>
+
+#include <fun4all/Fun4AllReturnCodes.h>
+
+#include <phool/PHCompositeNode.h>
+
+#include <TFile.h>
+#include <phool/getClass.h>
+#include "TH1.h"
+#include "TMath.h"
+//#include <jetbase/JetContainer.h>
+
+
+// Tower includes
+#include <calobase/RawCluster.h>
+#include <calobase/RawClusterContainer.h>
+#include <calobase/RawClusterUtility.h>
+#include <calobase/RawTower.h>
+#include <calobase/RawTowerContainer.h>
+#include <calobase/RawTowerGeom.h>
+#include <calobase/RawTowerGeomContainer.h>
+#include <calobase/TowerInfo.h>
+#include <calobase/TowerInfoContainer.h>
+#include <calobase/TowerInfoDefs.h>
+
+// Cluster includes
+#include <calobase/RawCluster.h>
+#include <calobase/RawClusterContainer.h>
+
+
+//____________________________________________________________________________..
+photonjetskinematics::photonjetskinematics(const std::string &outputfilename):
+SubsysReco("photonjetskinematics")
+, outfilename(outputfilename)
+{
+ std::cout << "photonjetskinematics::photonjetskinematics(const std::string &name, const std::string&outputfilename) Calling ctor" << std::endl;
+}
+//____________________________________________________________________________..
+photonjetskinematics::~photonjetskinematics()
+{
+  std::cout << "photonjetskinematics::~photonjetskinematics() Calling dtor" << std::endl;
+}
+
+//____________________________________________________________________________..
+int photonjetskinematics::Init(PHCompositeNode *topNode)
+{
+  std::cout << "photonjetskinematics::Init(PHCompositeNode *topNode) Initializing" << std::endl;
+  //outfile = new TFile(outfilename.c_str(), "RECREATE");
+  PHTFileServer::get().open(outfilename, "RECREATE");
+
```

Comment:
The following suggestion is for making sure the histogram names conform to the same pattern that most of the other modules follow: `h_<module name>_<what it is>_<additional tags>`. The first chunk just makes sure that the module name is all lower-case, and the second chunk systematically makes sure each histogram has the same prefix and suffix.

```suggestion

// make sure module name is lower case
std::string smallModuleName = modulename;
std::transform(
  smallModuleName.begin(),
  smallModuleName.end(),
  smallModuleName.begin(),
  ::tolower);

// construct histogram names
std::vector<std::string> vecHistNames = {
  "emcal_cluster_chi2",
  "emcal_cluster_energy",
  "emcal_cluster_eta_phi",
  "emcal_cluster_eta",
  "emcal_cluster_phi"};
for (auto& histName : vecHistName)
{
  histName.insert(0, "h_" + smallModuleName + "_");
  if (!histtag.empty())
  {
    histName.append("_" + histtag);
  }
}

```